deploy/terraform-datacommons-website/modules/apikeys/main.tf

```diff
@@ -62,20 +62,35 @@ gcloud alpha services api-keys create \
 --api-target=service=maps-backend.googleapis.com \
 --api-target=service=places-backend.googleapis.com
 
+EOT
+  }
+}
+
+resource "null_resource" "maps_api_key_fetch" {
+  # Regardless of the state, we always want to fetch the API key to a tmp file so
+  # the api key can be found in /tmp even in re-runs.
+  triggers = {
+    always_run = "${timestamp()}"
+  }
+
+  provisioner "local-exec" {
+    command = <<EOT
 touch /tmp/dc-website-api-key
 
 API_KEY_NAME=$(gcloud alpha services api-keys list --project=${var.project_id} --filter='displayName=maps-api-key${var.resource_suffix}' --format='value(name)' | head -n 1)
 gcloud alpha services api-keys get-key-string $API_KEY_NAME --format='value(keyString)' >> /tmp/dc-website-api-key
 
 EOT
   }
+
+  depends_on = [null_resource.maps_api_key]
```

Comment:
what is a "null_resource"?

Reply:
Added a readme!

https://github.com/datacommonsorg/website/blob/041f5b1f24b94b6b524ff8d8a00d88730f855e1c/deploy/terraform-datacommons-website/README.md#notes